packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,279 @@
+    setState(curState => ({
+      ...curState,
+      selectedValues: [],
+    }));
+    if (onClear) {
+      onClear();
+    }
+  }, [onClear]);
+
+  const keyDownHandler = useCallback(({ key }) => {
+    // setTimeout(0) needed when responding to key events to push back call
+    // to activeElement to after it is updated in the DOM
+    window.setTimeout(() => {
+      const focusedElement = document.activeElement;
+      switch (key) {
+        case 'Enter':
+          const match = focusedElement && focusedElement.id.match(`${state.id}-option-(.*)`);
+          if (match) {
+            handleSelect(match[1]);
+          }
+          break;
+        case 'ArrowUp':
+          if (focusedElement && focusedElement.id.match(`${state.id}-option-.*`)) {
+            const sibling = focusedElement.previousElementSibling as HTMLElement | null;
+            if (sibling) {
+              sibling.focus();
+            }
+          }
+          break;
+        case 'ArrowDown':
+          if (focusedElement && focusedElement.id === `${state.id}-valueContainer`) {
+            const optionsContainer = focusedElement.children[1];
+            if (optionsContainer) {
+              const toFocus = optionsContainer.children[0] as HTMLElement | undefined;
+              if (toFocus) {
+                toFocus.focus();
+              }
+            }
+          } else if (focusedElement && focusedElement.id.match(`${state.id}-option-.*`)) {
+            const sibling = focusedElement.nextElementSibling as HTMLElement | null;
+            if (sibling) {
+              sibling.focus();
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    }, 0);
+  }, [handleSelect, state.id]);
+
+  useEffect(() => {
+    window.removeEventListener('keydown', keyDownHandler);
+    window.addEventListener('keydown', keyDownHandler);
+    return () => {
+      window.removeEventListener('keydown', keyDownHandler);
+    };
+  }, [keyDownHandler]);
+
+  return (
+    <StyledContainer
+      data-testid={`${state.id}-valueContainer`}
+      elevation={elevation}
+      id={`${state.id}-valueContainer`}
+      name={name}
+      onBlur={handleBlur}
+      onFocus={(e: React.FocusEvent) => {
+        e.preventDefault();
+        setState(curState => ({ ...curState, isOpen: true }));
+      }}
+      tabIndex={tabIndex}
+    >
+      <Button
+        StyledContainer={StyledValueContainer}
+        onClick={(e) => e.preventDefault()}
+      >
+        <StyledValueItem>{((values.length && values) || state.selectedValues).join(', ')}</StyledValueItem>
+        <ValueIconContainer>
+          {clearable && (
+            <Icon
+              // TODO Fix issue with Icon not correctly exposing its onClick on its declaration file
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              onClick={handleClear}
+              onFocus={(e: FocusEvent) => e.stopPropagation()}
```

Comment:
The mdi docs say they don't want icon to have onclick for whatever reason - But you could wrap it in a div with those attached to it. (we probably want the padding around the icon to be clickable as well really)

Reply:
Interesting, I wonder why that's the case. I've moved it into a div.